backend/api/MQTT/MqttService.cs

```diff
@@ -0,0 +1,194 @@
+
+            string topic = messageReceivedEvent.ApplicationMessage.Topic;
+            var messageType = MqttTopics.TopicsToMessages.GetItemByTopic(topic);
+            if (messageType is null)
+            {
+                _logger.LogError("No message class defined for topic '{topicName}'", topic);
+                return;
+            }
+
+            switch (messageType)
+            {
+                case Type type when type == typeof(IsarMission):
+                    OnIsarTopicReceived<IsarMission>(content);
+                    break;
+                case Type type when type == typeof(IsarTask):
+                    OnIsarTopicReceived<IsarTask>(content);
+                    break;
+                case Type type when type == typeof(IsarStep):
+                    OnIsarTopicReceived<IsarStep>(content);
+                    break;
+                default:
+                    _logger.LogWarning(
+                        "No callback defined for MQTT message type '{type}'",
+                        messageType.Name
+                    );
+                    break;
+            }
+        }
+
+        private void OnConnected(MqttClientConnectedEventArgs obj)
+        {
+            _logger.LogInformation("Successfully connected.");
+        }
+
+        private void OnConnectingFailed(ManagedProcessFailedEventArgs obj)
+        {
+            _logger.LogWarning(
+                "Couldn't connect to MQTT broker. Exception: {message}\n"
+                    + "      Retrying in {time}s",
+                obj.Exception.Message,
+                _reconnectDelay.Seconds
+            );
+        }
+
+        private void OnDisconnected(MqttClientDisconnectedEventArgs obj)
+        {
+            _logger.LogInformation("Successfully disconnected.");
+        }
+
+        private void RegisterCallbacks()
+        {
+            _mqttClient.ConnectedHandler = new MqttClientConnectedHandlerDelegate(OnConnected);
+            _mqttClient.DisconnectedHandler = new MqttClientDisconnectedHandlerDelegate(
+                OnDisconnected
+            );
+            _mqttClient.ConnectingFailedHandler = new ConnectingFailedHandlerDelegate(
+                OnConnectingFailed
+            );
+            _mqttClient.ApplicationMessageReceivedHandler =
+                new MqttApplicationMessageReceivedHandlerDelegate(OnMessageReceived);
+        }
+
+        public void SubscribeToTopic(string topic)
+        {
+            var mqttSubscribeOptions = new MqttFactory()
+                .CreateSubscribeOptionsBuilder()
+                .WithTopicFilter(
+                    f =>
+                    {
+                        f.WithTopic(topic);
+                    }
+                )
+                .Build();
+            _mqttClient.SubscribeAsync(mqttSubscribeOptions.TopicFilters);
+        }
+
+        private void OnIsarTopicReceived<T>(string content) where T : MqttMessage
```

Comment:
Could "T" be spelled out for more readability?

Reply:
T is the standard C# convention for the "Type" in generic methods, I think it's best to follow this